contracts/andromeda_gumball/src/contract.rs

```diff
@@ -0,0 +1,238 @@
+        })?;
+
+    require(
+        payment.amount >= config.price,
+        ContractError::InsufficientFunds {},
+    )?;
+
+    //get randomed token_id
+    let mut all_token_ids = vec![];
+    let token_addr = deps.api.addr_humanize(&config.token_addr)?;
+    let mut start_after = "".to_string();
+
+    loop {
+        let tokens_res = query_tokens(
+            deps.querier,
+            token_addr.clone(),
+            info.sender.clone(),
+            Some(start_after.clone()),
+        )?;
+        let len = tokens_res.tokens.len();
+        for token_id in tokens_res.tokens {
+            all_token_ids.push(token_id.clone());
+            start_after = token_id;
+        }
+
+        if len < MAX_LIMIT as usize {
+            break;
+        }
+    }
+
+    let all_token_len = all_token_ids.len() as u64;
+    require(all_token_len > 0, ContractError::InsufficientTokens {})?;
+    // let selected_id = env.block.time.seconds() % all_token_len;
+    let selected_id = get_random_number(
+        deps.querier,
+        deps.api.addr_humanize(&config.terrand)?.to_string(),
+        all_token_len,
+    )?;
+    let token_id = all_token_ids[selected_id as usize].clone();
+
+    //transfer NFT
+    let msg = CosmosMsg::Wasm(WasmMsg::Execute {
+        contract_addr: deps.api.addr_humanize(&config.token_addr)?.to_string(),
+        msg: to_binary(&Cw721ExecuteMsg::TransferNft {
+            recipient: info.sender.to_string(),
+            token_id: token_id.clone(),
+        })?,
+        funds: vec![],
+    });
+
+    Ok(Response::new().add_message(msg).add_attributes(vec![
+        attr("action", "claim"),
+        attr("receiver", info.sender.to_string()),
+        attr("token_id", token_id),
+    ]))
+}
+
+fn execute_withdraw_funds(
+    deps: DepsMut,
+    env: Env,
+    info: MessageInfo,
+) -> Result<Response, ContractError> {
+    let config = CONFIG.load(deps.storage)?;
+    require(
+        is_contract_owner(deps.storage, info.sender.as_str())?,
+        ContractError::Unauthorized {},
+    )?;
+
+    let amount = query_balance(
+        &deps.querier,
+        env.contract.address,
+        config.stable_denom.clone(),
+    )?;
+    require(!amount.is_zero(), ContractError::InsufficientFunds {})?;
+    let msg = CosmosMsg::Bank(BankMsg::Send {
+        to_address: info.sender.to_string(),
+        amount: vec![coin(amount.u128(), config.stable_denom)],
+    });
+
+    Ok(Response::new().add_message(msg).add_attributes(vec![
+        attr("action", "withdraw_funds"),
+        attr("receiver", info.sender.to_string()),
+        attr("amount", amount),
+    ]))
+}
+
+pub fn get_random_number(
+    querier: QuerierWrapper,
+    terrand_addr: String,
+    max_count: u64,
+) -> StdResult<u64> {
+    let res: LatestRandomResponse = querier.query(&QueryRequest::Wasm(WasmQuery::Smart {
+        contract_addr: terrand_addr,
+        msg: to_binary(&TerrandQueryMsg::LatestDrand {})?,
+    }))?;
```

Comment:
Is this different for each transaction? If not couldn't it be gamed by waiting to mint until a particular number of tokens have been minted and then minting (since it looks like `max_count` might be the only variable changing).